src/pages/form-validation/password/index.tsx

```diff
@@ -0,0 +1,472 @@
+
+    const validatePasswordCriteria = useCallback((): void => {
+        if (newPasswordError) {
+            validateNewPassword();
+        } else {
+            setNewPasswordError(null);
+        }
+    }, [newPasswordError, validateNewPassword, setNewPasswordError]);
+
+    const onCurrentPasswordChange: OnChangeHandler = useCallback(
+        (event) => {
+            setCurrentPassword(event.target.value);
+            if (currentPasswordError) {
+                validateCurrentPassword();
+            } else {
+                setCurrentPasswordError(null);
+            }
+        },
+        [currentPasswordError, validateCurrentPassword, confirmPassword]
+    );
+
+    const onNewPasswordChange: OnChangeHandler = useCallback(
+        (event) => {
+            setNewPassword(event.target.value);
+            setConfirmPasswordError('');
+            validatePasswordCriteria();
+            if (event.target.value !== confirmPassword && confirmPassword !== '') {
+                setConfirmPasswordError(PASSWORD_MISMATCH);
+            }
+        },
+        [setNewPassword, validatePasswordCriteria, confirmPassword]
+    );
+
+    const onConfirmPasswordChange: OnChangeHandler = useCallback(
+        (event) => {
+            setConfirmPasswordError('');
+            setConfirmPassword(event.target.value);
+            if (newPassword !== event.target.value) {
+                setConfirmPasswordError(PASSWORD_MISMATCH);
+            }
+        },
+        [newPassword]
+    );
+
+    useEffect(() => {
+        setPasswordErrors({
+            minLengthRequired: newPassword.length >= 8,
+            atLeast1UpperCharRequired: upperCharRegex.test(newPassword),
+            atLeast1LowerCharRequired: lowerCharRegex.test(newPassword),
+            atLeast1NumberRequired: numberRegex.test(newPassword),
+            atLeast1SplCharRequired: splCharRegex.test(newPassword),
+        });
+    }, [newPassword]);
+
+    const meetsRequirements = useCallback(
+        (): boolean =>
+            Boolean(passwordErrors.atLeast1LowerCharRequired) &&
+            Boolean(passwordErrors.atLeast1NumberRequired) &&
+            Boolean(passwordErrors.atLeast1SplCharRequired) &&
+            Boolean(passwordErrors.minLengthRequired) &&
+            Boolean(passwordErrors.atLeast1UpperCharRequired),
+        [passwordErrors]
+    );
+
+    const submitEnabled = useCallback(
+        (): boolean => meetsRequirements() && newPassword === confirmPassword && Boolean(currentPassword),
+        [meetsRequirements, newPassword, confirmPassword, currentPassword]
+    );
+
+    const clearForms = (): void => {
+        setCurrentPassword('');
+        setConfirmPassword('');
+        setNewPassword('');
+    };
+
+    const passwordHintText = (error: boolean): string => (error ? theme.palette.text.primary : Colors.gray[200]);
+
+    return (
+        <div
+            style={{
+                backgroundColor: '#E5E5E5',
```

Comment:
theme.palette.background.default

Reply:
Ok I'm now using the theme palette. Good catch.